tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/debugger_component.ts

```diff
@@ -12,18 +12,41 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-import {Component, Input} from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
 import {DebuggerRunListing} from './store/debugger_types';
+import {requestExecutionDigests} from './actions';
 
 @Component({
   selector: 'debugger-component',
   templateUrl: './debugger_component.ng.html',
   styleUrls: ['./debugger_component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class DebuggerComponent {
+export class DebuggerComponent implements OnChanges {
   @Input()
   runs: DebuggerRunListing = {};
 
   @Input()
   runIds: string[] = [];
+
+  @Input()
+  activeRunId: string | null = null;
+
+  @Output()
+  onActiveRunIdChange = new EventEmitter<string>();
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['activeRunId'] && changes['activeRunId'].currentValue) {
+      this.onActiveRunIdChange.emit(changes['activeRunId']
```

Comment:
This feels very wrong. The data that comes from store is dispatched back to the flux. Why do you need to do this?